dot/network/config.go

```diff
@@ -161,6 +166,19 @@ func (c *Config) checkState() (err error) {
 // service, if a key does not exist or cannot be loaded, it creates a new key
 // using the random seed (if random seed is not set, creates new random key)
 func (c *Config) buildIdentity() error {
+	if c.NodeKey != "" {
+		privateKeySeed, err := common.HexToBytes("0x" + c.NodeKey)
+		if err != nil {
+			return fmt.Errorf("parsing hex encoding of ed25519 private key: %w", err)
+		}
+		key := ed25519.NewKeyFromSeed(privateKeySeed)
+		privateKey, err := crypto.UnmarshalEd25519PrivateKey(key)
+		if err != nil {
+			return fmt.Errorf("decoding ed25519 bytes: %w", err)
+		}
+		c.privateKey = privateKey
+		return nil
+	}
```

Comment:
supernit an empty line below would be nice

```suggestion
	}

```